conf/nginx/st2.conf

```diff
@@ -15,7 +15,9 @@ server {
   if ($ssl_protocol = "") {
        return 308 https://$host$request_uri;
   }
-
+  if ($request_method !~ ^(GET|HEAD|POST)$ ) {
+       return 405;
+  }
```

Comment:
Is this section required below for the main API server on 443 ssl?

Reply:
If we also restrict this on the 443 server, note these
- There are API functons that use `PUT`. Ref: https://github.com/StackStorm/st2/blob/master/st2api/st2api/controllers/v1/actions.py#L156
- There are API functions that use `DELETE` to remove resources. How will this impact these functions? Ref: https://github.com/StackStorm/st2/blob/master/st2api/st2api/controllers/v1/actions.py#L208

Reply:
Yeah, as @m4dcoder said this is likely untested and won't work - API operations also utilize PUT, POST, OPTIONS (for cors) operations.

Reply:
@m4dcoder - I have modified the code and the only TRACE will not be allowed.